src/python/system/shell.py

```diff
@@ -144,7 +144,7 @@ def clear_data_directories_on_low_disk_space():
 
 def clear_device_temp_directories():
   """Clear device specific temp directories."""
-  if environment.is_android():
+  if environment.is_android() and not environment.is_android_emulator():
```

Comment:
I dont understand this, why won't we want to clear temp dirs on android emulator, we might not be spinning up emulator on every task. same for clear_testcase_directory below.

Reply:
I added this because it can be called before the device is started (commands.py:382) which will cause errors. Instead, I can more directly check for the emulator being started or the device being available by checking for the ANDROID_SERIAL variable being set.

Reply:
Ok!